firebase-database/src/main/java/com/google/firebase/database/FirebaseDatabase.java

```diff
@@ -302,9 +294,23 @@ public synchronized void setPersistenceCacheSizeBytes(long cacheSizeInBytes) {
     this.config.setPersistenceCacheSizeBytes(cacheSizeInBytes);
   }
 
-  @Nullable
-  private static EmulatedServiceSettings getEmulatorServiceSettings(@NonNull FirebaseApp app) {
-    return app.getEmulatorSettings().getServiceSettings(EMULATOR);
+  /**
+   * Modify this FirebaseDatabase instance to communicate with the Realtime Database emulator.
+   *
+   * <p>Note: this must be called before this instance has been used to do any database operations.
+   *
+   * @param host the emulator host (ex: 10.0.2.2)
+   * @param port the emulator port (ex: 9000)
+   */
+  public void useEmulator(@NonNull String host, int port) {
+    if (this.repo != null) {
```

Comment:
What is the expected behavior in the following scenario?

```kotlin
val custom = Firebase.database("http://localhost:8000")
readAndWriteTo(custom)
val default = Firebase.database.useEmulator("example.com", 123)
```

afaict this will cause custom to keep talking to localhost, while default will talk to the emulator.

Do we want to throw in this case?

Reply:
No we do not.  That was something we could have done in the old "hold it all in FirebaseApp" proposal but it's not longer a feasible edge case to handle.